pil-analyzer/src/evaluator.rs

```diff
@@ -445,24 +452,48 @@ impl<'a> Definitions<'a> {
                 .into()
             }
         } else {
-            match value {
-                Some(FunctionValueDefinition::Expression(TypedExpression {
-                    e: value,
-                    type_scheme,
-                })) => {
-                    let type_args = type_arg_mapping(type_scheme, type_args);
-                    evaluate_generic(value, &type_args, symbols)?
+            let impl_pos = type_args
+                .as_ref()
+                .and_then(|type_args| poly.resolved_impls.get(type_args).as_ref().copied());
+
+            match impl_pos {
+                Some(index) => {
+                    let fn_name = poly.name.rsplit("::").next().unwrap().to_string();
```

Comment:
Looks like we also need to store the function name.

Reply:
At best we could store a reference to the function directly (i.e. no impl index, no function name), do you think this can be done? We probably need a lifetime parameter and that could be quite a big change we don't want to do.

Reply:
Oh and it probably doesn't even work to assign that reference since we need a mut ref to the ast to do that but we store a ref to one of its children inside it. :(

Reply:
Another solution could be to sort the functions in the same way as we have them in the trait (and then only store an index and not a function name). Then the re-printing might swap them around but maybe that's ok?